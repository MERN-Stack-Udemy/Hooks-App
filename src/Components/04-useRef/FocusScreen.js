import React from 'react'

export const FocusScreen = () => {
  return (
    <div>
      <h2>Ficus Screen</h2>
    </div>
  )
}
