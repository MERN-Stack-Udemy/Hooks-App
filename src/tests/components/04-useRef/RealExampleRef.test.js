import { shallow } from "enzyme"
import { RealExampleRef } from "../../../Components/04-useRef/RealExampleRef"

describe('Test on <RealExampleRef />', () => {
  
  test('should display the component correctly', () => {
    const wrapper = shallow( <RealExampleRef /> );
    expect( wrapper.find('MultipleCustomHooks').exists() ).toBe( false );
    expect( wrapper ).toMatchSnapshot();
  })

  test('should display the <MultipleCustomHooks/> ', () => {
    const wrapper = shallow( <RealExampleRef /> );
    wrapper.find('button').simulate('click');
    expect( wrapper.find('MultipleCustomHooks').exists() ).toBe( true );
    expect( wrapper ).toMatchSnapshot();
  })
  
  
})
