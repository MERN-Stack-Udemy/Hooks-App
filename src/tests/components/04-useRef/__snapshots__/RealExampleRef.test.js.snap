// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <RealExampleRef /> should display the <MultipleCustomHooks/>  1`] = `
<div>
  <h1>
    Real Example Ref
  </h1>
  <hr />
  <MultipleCustomHooks />
  <button
    className="btn btn-primary mt-2"
    onClick={[Function]}
  >
    Hide
  </button>
</div>
`;

exports[`Test on <RealExampleRef /> should display the component correctly 1`] = `
<div>
  <h1>
    Real Example Ref
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-2"
    onClick={[Function]}
  >
    Show
  </button>
</div>
`;
